src/Lecture2.hs

```diff
@@ -62,7 +80,7 @@ lazyProduct = error "TODO"
 "ccaabb"
 -}
 duplicate :: [a] -> [a]
-duplicate = error "TODO"
+duplicate = concat . fmap (take 2 . repeat)
```

Comment:
That's a correct solution 👍🏻 You can even use the standard function `concatMap` to replace `concat . fmap` and `replicate 2` to replace `take 2. repeat`.

However, even in that case, the implementation allocates temporary lists inside and appends them with `++`.

It is more efficient to construct lists using the cons operator (`:`) directly without introducing intermediate values or adding lists with `(++)` whenever possible.